cherryontech/jupiter-jumpers
# Updating UI and behavior after registering

## Description

Updates to behavior and UI after user registration:
- redirected to home page after account created
- Welcome message modal appears after user authenticates for the first time (creating account)
- Modal does not appear again after it is closed out 
- user context created so logged in user information is global
- firebase API functions for user added to new api file
- new field closedConfirmationMessage added to DB to track whether user has closed message or not 

## Related Issue

https://cherryontech-jupiter-jumpers.atlassian.net/browse/SCRUM-12
https://cherryontech-jupiter-jumpers.atlassian.net/browse/SCRUM-13

## Related Story Card

_Provide a link to the relevant story card._

## Type of Changes

Feature update.

## Acceptance Criteria

- [x] User is taken to home page after creating account 
- [x] The confirmation modal appears after creating account 
- [x] The user can click 'get started' to close out the modal
- [x] After refreshing, the user remains logged in
- [x] The confirmation modal does not appear again after it is closed 

## Update Screenshots




### Before
This was the view pre and post authentication 
<img width="1439" alt="Screenshot 2024-04-27 at 5 22 44 PM" src="https://github.com/cherryontech/jupiter-jumpers/assets/46732901/eef96df3-5f2a-474f-aa1b-83509ca9f0ce">


### After

Modal appears after first authentication
![Screenshot 2024-04-27 at 5 13 38 PM](https://github.com/cherryontech/jupiter-jumpers/assets/46732901/8e10447d-d85f-4c04-aec6-6d3589c24229)

UI reflects logged in state
![Screenshot 2024-04-27 at 5 13 46 PM](https://github.com/cherryontech/jupiter-jumpers/assets/46732901/5b614cc1-4d45-4305-a9f8-68ab9b09332e)

## Testing Instructions

1. Create an account
2. Make sure app directs you to home page (/)
3. View confirmation message modal
4. Close out modal
5. Make sure app reflects logged in state
6. Refresh to make sure you remain logged in (this takes a second so the logged out state will flash initially)

## Learnings (Optional)

_Document any things you learned or 'gotchas' you experienced._
